src/xform_to_datalog/arcs_manifest_tree/particle_spec.h

```diff
@@ -135,11 +135,11 @@ class ParticleSpec {
   // A map of HandleConnectionSpec names to HandleConnectionSpecs.
   absl::flat_hash_map<std::string, raksha::ir::HandleConnectionSpec>
     handle_connection_specs_;
-  // The PredicateDecoder object which was used to construct the predicates
+  // The PredicateArena object which was used to construct the predicates
   // on all of the checks on this ParticleSpec. This also owns those
   // predicates, so holding this field here causes the checks to live as long
   // as the ParticleSpec.
-  raksha::ir::proto::PredicateDecoder predicate_decoder_;
+  std::unique_ptr<raksha::ir::PredicateArena> predicate_arena_;
```

Comment:
It doesn't feel quite right to have an object just for the sole purpose of owning these predicates. There should be a way to avoid such work around. Once the protos and ir is detangled, we should look into cleaning up these structures.

Reply:
Definitely. This is very much a consequence of the lack of class-level distinction between spec and impl object versions, and would go away if we fix that issue.